perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -518,6 +643,240 @@ def SingleStreamThroughputBenchmark(storage_schema, host_to_connect=None):
+        object_records.append([name, size])
+    if os.path.exists(FLAGS.objects_written_file):
+      os.remove(FLAGS.objects_written_file)
+    with open(FLAGS.objects_written_file, 'w') as out:
+      json.dump(object_records, out)
+
+  # operation_records is the data we send back to the controller.
+  operation_records = []
+  for result in results:
+    operation = result['operation']
+    stream_num = result['stream_num']
+    for start_time, latency, size in zip(result['start_times'],
+                                         result['latencies'],
+                                         result['sizes']):
+      operation_records.append({'operation': operation,
+                                'start_time': start_time,
+                                'latency': latency,
+                                'size': size,
+                                'stream_num': stream_num})
+  json.dump(operation_records, sys.stdout, indent=0)
+
+
+def MultiStreamReads(storage_schema, host_to_connect=None):
+  """Run multi-stream read threads."""
+
+  # Read the object records that the MultiStreamWriter left for us.
+  if FLAGS.objects_written_file is None:
+    raise ValueError('The MultiStreamRead benchmark needs a list of object '
+                     'names to read from. Use '
+                     '--objects_written_file=<filename>.')
+  with open(FLAGS.objects_written_file, 'r') as object_file:
+    object_records = json.load(object_file)
+
+  num_workers = FLAGS.num_streams
+  objects_by_worker = []
+  objects_per_worker = (len(object_records) + num_workers - 1) // num_workers
+  for i in xrange(num_workers):
+    # Do a simple, even distribution of objects to workers.
+    low = i * objects_per_worker
+    high = min((i + 1) * objects_per_worker, len(object_records))
+    objects_by_worker.append(object_records[low:high])
+
+  results = RunThreadedWorkers(
+      ReadWorker,
+      (storage_schema,
+       host_to_connect,
+       FLAGS.start_time),
+      per_thread_args=objects_by_worker)
+
+  # send data back to the controller
+  operation_records = []
+  for result in results:
+    operation = result['operation']
+    stream_num = result['stream_num']
+    for start_time, latency, size in zip(result['start_times'],
+                                         result['latencies'],
+                                         result['sizes']):
+      operation_records.append({'operation': operation,
+                                'stream_num': stream_num,
+                                'start_time': start_time,
+                                'latency': latency,
+                                'size': size})
+  json.dump(operation_records, sys.stdout, indent=0)
+
+
+def SleepUntilTime(when):
+  """Sleep until a given time.
+
+  Args:
+    when: float. The time to sleep to, as a POSIX timestamp.
+  """
+
+  now = time.time()
+  sleep_time = when - now
+  if sleep_time > 0.0:
```

Comment:
sleep_time being 0 should be supported, if sleep time is 0, just skip sleep instead of logging a warning here.